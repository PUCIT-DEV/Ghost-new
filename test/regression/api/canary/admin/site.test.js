const chai = require('chai');
const {expect} = require('chai');
const {any, stringMatching} = require('expect');
const chaiJestSnapshot = require('@ethanresnick/chai-jest-snapshot');
const framework = require('../../../../utils/e2e-framework');
const config = require('../../../../../core/shared/config');

chai.use(chaiJestSnapshot);

describe('Config API', function () {
    let request;

    before(async function () {
        chaiJestSnapshot.resetSnapshotRegistry();
        request = await framework.getAgent('/ghost/api/canary/admin/');
    });

    beforeEach(function () {
        chaiJestSnapshot.configureUsingMochaContext(this);
    });

    it('can retrieve config and all expected properties', async function () {
        const res = await request
            .get('site/')
            .set('Origin', config.get('url'));

        expect(res.body.site).to.matchSnapshot({
            version: stringMatching(/\d+\.\d+/)
        });
        expect(res.headers).to.matchSnapshot({
            date: any(String),
            etag: any(String)
        });
    });
});
