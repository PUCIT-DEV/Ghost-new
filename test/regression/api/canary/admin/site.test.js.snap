// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config API can retrieve config and all expected properties 1`] = `
Object {
  "accent_color": "#FF1A75",
  "description": "Thoughts, stories and ideas",
  "icon": null,
  "logo": null,
  "title": "Ghost",
  "url": "http://127.0.0.1:2369/",
  "version": StringMatching /\\\\d\\+\\\\\\.\\\\d\\+/,
}
`;

exports[`Config API can retrieve config and all expected properties 2`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "connection": "close",
  "content-length": "167",
  "content-type": "application/json; charset=utf-8",
  "date": Any<String>,
  "etag": Any<String>,
  "vary": "Origin, Accept-Encoding",
  "x-powered-by": "Express",
}
`;
